### PR TITLE
Main branch code review

### PR DESCRIPTION
**Pull request by StuartCowley**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Virtual Pet
+
+A simple virtual pet program created in JavaScript.
+<br><br>
```
Comment (StuartCowley) [inline README.md:4]: You don't need to add `<br>` tags in markdown language, just press enter a few times :)

File: README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+If Bobbie gets to 30 days old they will DIE 😢
+<br>
+<br>
+
+## Contributing
+
+Pull requests are welcome. For major changes, please open an issue first to discuss what you would like to change.
+
+Please make sure to update tests as appropriate.
```
Comment (StuartCowley) [inline README.md:36]: Nice touch

File: src/pet.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
+const initialAge = 0;
```
Comment (StuartCowley) [inline src/pet.js:1]: This is nicely laid out and easy to understand, nicely done

File: __tests__/pet.test.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
+const Pet = require("../src/pet");
+
+describe("constructor", () => {
+  const pet = new Pet("Fido");
+  it("returns an object", () => {
+    expect(new Pet()).toBeInstanceOf(Object);
+  });
+  it("renames pet", () => {
```
Comment (StuartCowley) [inline __tests__/pet.test.js:8]: This is just naming the pet surely?

File: __tests__/pet.test.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
+const Pet = require("../src/pet");
+
+describe("constructor", () => {
+  const pet = new Pet("Fido");
+  it("returns an object", () => {
```
Comment (StuartCowley) [inline __tests__/pet.test.js:5]: It would be good to separate each `it` block by a clear line to make it a bit easier to read

File: __tests__/pet.test.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    expect(new Pet()).toBeInstanceOf(Object);
+  });
+  it("renames pet", () => {
+    expect(pet.name).toBe("Fido");
+  });
+  it("has an initial age of 0", () => {
```
Comment (StuartCowley) [inline __tests__/pet.test.js:11]: For these values, would it be better to define them in some variables at the top so they can be referenced throughout the file in the same way as in `pet.js`?

File: __tests__/pet.test.js [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    expect(pet.hunger).toBe(0);
+  });
+  it("has an intial fitness of 10", () => {
+    expect(pet.fitness).toBe(10);
+  });
+});
+
+describe("growUp", () => {
+  const pet = new Pet("Fido");
+  it("thows an error if pet is not alive", () => {
```
Comment (StuartCowley) [inline __tests__/pet.test.js:24]:
```suggestion
  it("throws an error if pet is not alive", () => {
```

File: src/pet.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    return (
+      this.fitness > deathFitness &&
+      this.hunger < deathHunger &&
+      this.age < deathAge
+    );
+  }
+
+  growUp() {
+    if (!this.isAlive) {
+      throw new Error("Your pet is no longer alive :(");
```
Comment (StuartCowley) [inline src/pet.js:33]: This message is referenced several times in the file, could it be put into a constant along with the values?

File: __tests__/pet.test.js [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+});
+
+describe("walk", () => {
+  const pet = new Pet("Fido");
+  it("increases fitness level", () => {
+    pet.fitness = 4;
+    pet.walk();
+    expect(pet.walk).toBeInstanceOf(Function);
+    expect(pet.fitness).toBeLessThanOrEqual(10);
+    expect(pet.fitness).toBeGreaterThanOrEqual(0);
```
Comment (StuartCowley) [inline __tests__/pet.test.js:44]: If youre checking for a specific value as on line 45, why check for 2 other ranges like this?

File: __tests__/pet.test.js [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+    expect(pet.fitness).toEqual(8);
+  });
+  it("thows an error if pet is not alive", () => {
```
Comment (StuartCowley) [inline __tests__/pet.test.js:47]:
```suggestion
  it("throws an error if pet is not alive", () => {
```

File: __tests__/pet.test.js [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+    pet.fitness = 0;
+    expect(() => pet.walk()).toThrow("Your pet is no longer alive :(");
+  });
+});
+
+describe("feed", () => {
+  const pet = new Pet("Fido");
+  it("thows an error if pet is not alive", () => {
```
Comment (StuartCowley) [inline __tests__/pet.test.js:55]:
```suggestion
  it("throws an error if pet is not alive", () => {
```

File: __tests__/pet.test.js [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+    pet.feed();
+    expect(pet.feed).toBeInstanceOf(Function);
+    expect(pet.hunger).toBeGreaterThanOrEqual(0);
+    expect(pet.hunger).toEqual(0);
+  });
+});
+
+describe("checkUp", () => {
+  const pet = new Pet("Fido");
+  it("returns death string if pet is not alive", () => {
```
Comment (StuartCowley) [inline __tests__/pet.test.js:71]: death string 😂  brutal

File: __tests__/pet.test.js [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+    pet.growUp();
+    expect(pet.age).toEqual(1);
+    expect(pet.hunger).toEqual(5);
+    expect(pet.fitness).toEqual(7);
+  });
+});
+
+describe("walk", () => {
+  const pet = new Pet("Fido");
+  it("increases fitness level", () => {
```
Comment (StuartCowley) [inline __tests__/pet.test.js:39]: Another test could be 'fitness cant go above initial fitness'

File: __tests__/pet.test.js [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+  });
+});
+
+describe("feed", () => {
+  const pet = new Pet("Fido");
+  it("thows an error if pet is not alive", () => {
+    pet.age = 30;
+    expect(() => pet.feed()).toThrow("Your pet is no longer alive :(");
+  });
+  it("decreases hunger level", () => {
```
Comment (StuartCowley) [inline __tests__/pet.test.js:59]: Could add another test to see if pet dies if it goes below minimum hunger level

**Review by StuartCowley — COMMENTED** [review]
Good clean code here, nicely done! Have left a few suggestions around readability, a couple of other test suggestions, using more constants in tests and a few typos but thats about it
